ERCS/erc-7945.md

```diff
@@ -0,0 +1,250 @@
+
+Note:
+
+* Callers MUST handle `false` from returns (bool success). Callers MUST NOT assume that `false` is never returned!
+* Implementations can fully customize the proof system, (de)serialization strategies of `bytes` and/or the business workflow. For example, when implementing "Zether" confidential token contracts, the `_confidentialValue` and accounts' confidential balances will be encrypted homomorphically under ElGamal public keys, and `_proof` will consist 3 parts to check:
+
+  * `_confidentialValue` is well encrypted under public keys of caller's and `_spender` 's;
+  * The plaintext `value` behind `_confidentialValue` is non-negative;
+  * The caller's confidential balance is actually enough to pay the plaintext `value` behind `_confidentialValue` .
+
+```Solidity
+function confidentialApprove(
+  address _spender,
+  bytes memory _confidentialValue, 
+  bytes memory _proof
+) public returns (bool success)
+```
+
+##### confidentialAllowance
+
+Returns the allowance part which `_spender` is still allowed to withdraw from `_owner` .
+
+```Solidity
+functinon confidentialAllowance(address _owner, address _spender)
+public view returns (bytes memory _confidentialValue)
+```
+
+#### Events
+
+##### ConfidentialTransfer
+
+MUST trigger when tokens are transferred.
+
+Specifically, if tokens are transffered through function `confidentialTransferFrom` , `_spender` address MUST be set to caller's; otherwise, it SHOULD be set to `0x0` .
+
+A confidential token contract,
+
+* which creates new tokens SHOULD trigger a `ConfidentialTransfer` with the `_from` address set to `0x0` when tokens are minted;
+* which destroys existent tokens SHOULD trigger a `ConfidentialTransfer` with the `_to` address set to `0x0` when tokens are burnt.
+
+```Solidity
+event ConfidentialTransfer(
+  address indexed _spender,
+  address indexed _from, 
+  address indexed _to, 
+  bytes _confidentialTransferValue
+)
+```
+
+##### ConfidentialApproval
+
+MUST trigger on any successful call to `confidentialApprove(address,bytes,bytes)` .
+
+```Solidity
+event ConfidentialApproval(
+  address indexed _owner,
+  address indexed _spender,
+  bytes _currentAllowancePart,
+  bytes _allowancePart
+)
+```
+
+## Rationale
+
+
+Confidential transactions have been implemented in many blockchains, either natively through blockchain protocols like Monero and Zcash, or through smart contracts like Zether (DOI: 10.1007/978-3-030-51280-4_23) without modifying blockchain protocol.
+
+However, when it comes to the latter way, actually no standards are proposed yet to illustrate such contracts. Users and applications cannot easily detect whether a token contract supports confidential transactions or not, and so hardly can they make transfers without revealing the actual amount.
+
+Consequently, this proposal is to standardize the confidential transaction supported token contracts, meanwhile without loss of generality, by only specifying core methods and events.
```

Comment:
This content is describing the proposal as a whole, and so belongs in the rationale section. My pet example for the distinction between motivation and rationale is:

> Motivation: We need to build a shed because...
> Rationale: We chose to paint the shed red because...